lib/batch_verification/src/sequencer_component.rs

```diff
@@ -0,0 +1,372 @@
+use crate::config::BatchVerificationConfig;
+use crate::{BatchVerificationRequestError, BatchVerificationServer};
+use crate::{BatchVerificationResponse, BatchVerificationResult};
+use alloy::primitives::Address;
+use async_trait::async_trait;
+use dashmap::DashMap;
+use futures::FutureExt;
+use futures::future::join_all;
+use std::sync::Arc;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::time::Duration;
+use tokio::sync::mpsc::{self, Sender};
+use tokio::time::Instant;
+use zksync_os_l1_sender::batcher_metrics::BatchExecutionStage;
+use zksync_os_l1_sender::batcher_model::{
+    BatchForSigning, BatchSignatureData, SignedBatchEnvelope,
+};
+use zksync_os_observability::{ComponentStateReporter, GenericComponentState};
+use zksync_os_pipeline::{PeekableReceiver, PipelineComponent};
+use zksync_os_types::BatchSignatureSet;
+
+fn report_exit<T, E: std::fmt::Debug>(name: &'static str) -> impl Fn(Result<T, E>) {
+    move |result| match result {
+        Ok(_) => tracing::warn!("{name} unexpectedly exited"),
+        Err(e) => tracing::error!("{name} failed: {e:#?}"),
+    }
+}
+pub struct BatchVerificationPipelineStep<E> {
+    config: BatchVerificationConfig,
+    _phantom: std::marker::PhantomData<E>,
+}
+
+impl<E> BatchVerificationPipelineStep<E> {
+    pub fn new(config: BatchVerificationConfig) -> Self {
+        Self {
+            config,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+}
+
+#[async_trait]
+impl<E: Send + Sync + 'static> PipelineComponent for BatchVerificationPipelineStep<E> {
+    type Input = BatchForSigning<E>;
+    type Output = SignedBatchEnvelope<E>;
+
+    const NAME: &'static str = "batch_verification";
+    const OUTPUT_BUFFER_SIZE: usize = 5;
+
+    async fn run(
+        self,
+        mut input: PeekableReceiver<Self::Input>,
+        output: mpsc::Sender<Self::Output>,
+    ) -> anyhow::Result<()> {
+        if self.config.enabled {
+            let (server, response_receiver) = BatchVerificationServer::new();
+            let server = Arc::new(server);
+            let response_channels = Arc::new(DashMap::new());
+
+            let server_for_fut = server.clone();
+            let server_address = self.config.address.clone();
+            let server_fut = async move { server_for_fut.run_server(server_address).await }
```

Comment:
Can't we just have `server_for_fut.run_server(server_address).boxed()`? Why do we need to have `async move { <x>.await }`? Here and below